web-quiz-app/src/pages/JoinQuiz.jsx

```diff
@@ -1,23 +1,46 @@
 import HeaderStudent from "../components/HeaderStudent";
 import './css/QuizControl.css';
 import React from "react";
+import { useEffect, useState } from 'react';
+import QuizDataService from "../services/quiz.service";
 
 function JoinQuiz() {
+    const [nextPage, setNextPage] = useState(false)
+    const [quizID, setQuizID] = useState();
+
+    useEffect(() => {
+        if(nextPage) {
+            window.location.href = '/quizAnswer/' + document.getElementById("quizID").value;
+        }
+        else {
+            alert("This quiz doesn't exost, or is inaccessible")    
+        }
+    }, [nextPage])
+
     return (
         <div>
             <HeaderStudent></HeaderStudent>
             <div class="form-group">
                 <label for="name">Quiz ID</label>
-                <input type="text" class="form-control" id="quizID" placeholder="Enter Quiz ID Here"></input>
+                <input type="text" class="form-control" id="quizID" placeholder="Enter Quiz ID Here" onChange={event => setQuizID(event.target.value)}></input>
                 <button class="btn btn-primary" onClick={NavigateToQuiz}>Join</button>
             </div>
         </div>
     );
-}
-
-function NavigateToQuiz() {
-    window.location.href = '/quizAnswer/' + document.getElementById("quizID").value;
 
+    function NavigateToQuiz() {
```

Comment:
If I understand correctly, you moved this function within the JoinQuiz function. Would you need to update it on line 26? I know you do for classes, but wasn't sure if it was necessary in this case. If it works, then please disregard this comment.